central/config/datastore/datastore.go

```diff
@@ -42,6 +56,54 @@ var (
 
 type datastoreImpl struct {
 	store store.Store
+
+	publicConfigCache *expirable.LRU[string, *storage.PublicConfig]
+}
+
+// GetCachedPublicConfig returns the public part of the Central config.
+// The primary data source will be the cache and the secondary the database.
+func (d *datastoreImpl) GetCachedPublicConfig(ctx context.Context) (*storage.PublicConfig, error) {
```

Comment:
What is the motivation (and hence value) for adding caching for public config?

Reply:
The public config is served from an anonymous / public endpoint.
The goal of caching that data is to shield the database in case of request flooding.

Reply:
Comments added to document the considerations around public config caching (one detailed and shorter references to that detailed comment).